libs/wire-subsystems/test/unit/Wire/MockInterpreters/UserGroupStore.hs

```diff
@@ -0,0 +1,83 @@
+{-# OPTIONS_GHC -Wno-ambiguous-fields #-}
+
+-- FUTUREWORK: move this next to Postgres interpreter; write integration tests that run random,
+-- valid command sequences against both and compare.
+
+module Wire.MockInterpreters.UserGroupStore where
+
+import Data.Default
+import Data.Id
+import Data.Json.Util
+import Data.Map qualified as Map
+import GHC.Stack
+import Imports
+import Polysemy
+import Polysemy.Internal (Append)
+import Polysemy.State
+import System.Random (StdGen, mkStdGen)
+import Wire.API.User
+import Wire.API.UserGroup
+import Wire.MockInterpreters.Random
+import Wire.Sem.Random qualified as Rnd
+import Wire.UserGroupStore
+
+data UserGroupInMemState = UserGroupInMemState
+  { userGroups :: Map (TeamId, UserGroupId) UserGroup,
+    now :: UTCTimeMillis
```

Comment:
We should use the `Wire.Sem.Now` for this. That one has a test interpreter which allows time modifications. So we don't have to re-implement this in every store effect which will use time.

Reply:
sorry, oversight.

Reply:
[ff95607](https://github.com/wireapp/wire-server/pull/4545/commits/ff95607a1c416bce6b09df0b22ae54058e54a732)